components/dashboard/src/index.tsx

```diff
@@ -1,18 +1,18 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
-import "./tailwind.output.css"
 import App from './App';
-import { ServiceContext, SimpleServiceImpl } from './service/service';
+import { AppProvider } from './contexts';
+import { service } from './service/service';
+
+import "./tailwind.output.css"
+
+service.getOrLoadUser().then(user => console.log(user.name));
 
-const service = new SimpleServiceImpl();
-service.service.server.getLoggedInUser().then(user => {
-  service.user = user;
-  ReactDOM.render(
+ReactDOM.render(
     <React.StrictMode>
-      <ServiceContext.Provider value={service}>
-        <App />
-      </ServiceContext.Provider>
+        <AppProvider>
```

Comment:
We could pass in the `userPromise` as a prop. Seems like this is the only client for that. So no need to store it globally.